### PR TITLE
[MRG] DOC: misc doc improvements

### PR DESCRIPTION
**Pull request by jasmainak**

### CODE REVIEW

**Comment by jasmainak** [comment]
The CircleCI failure seems to be spurious. I hope it's fixed when I rebuild it. What I've done here is:

- add intersphinx so we can cross-link to documentation of other repositories like MNE
- order the examples from basic to more advanced.
- add links to API documentation in the examples

I just wanted to show what all can be done with sphinx/sphinx-gallery. I suggest doing a fast merge and building on top of this @cjayb @ntolley @rythorpe

File: examples/plot_firing_pattern.py [hunk 1]
```diff
@@ -90,8 +107,8 @@
 plt.show()
 
 ###############################################################################
-# Let's plot the soma voltage along with the spiking activity with raster
-# plots and histograms for the Pyramidal cells.
+# Finally, we can plot the soma voltage along with the spiking activity with
+# raster plots and histograms for the Pyramidal cells.
```
Comment (rythorpe) [inline examples/plot_firing_pattern.py:111]: Nitpick: is "Pyramidal cells" here supposed to link to the source code? If not, it probably doesn't need an uppercase "P".

File: doc/conf.py [hunk 2]
```diff
@@ -177,6 +180,15 @@
...
 
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3', None),
+    'mne': ('https://mne.tools/dev', None),
+    'numpy': ('https://numpy.org/devdocs', None),
+    'scipy': ('https://scipy.github.io/devdocs', None),
+    'matplotlib': ('https://matplotlib.org', None)
+}
+intersphinx_timeout = 5
+
```
Comment (rythorpe) [inline doc/conf.py:191]: So all of the external links to other projects are done with intersphinx? I'm just trying to understand better how this all works....

Comment (jasmainak) [reply doc/conf.py:191]: Just the links to the API pages.

Comment (jasmainak) [reply doc/conf.py:191]: e.g., the link here to plt.subplots:

image

**Comment by rythorpe** [comment]
Everything looks good to me. Are we good to merge once the CIs turn green or do you want to wait for another review @jasmainak ?

**Comment by jasmainak** [comment]
I would say go for merge since these are pretty minor edits. You can always change something in your own PR :)

**Pull request merged**